src/Microsoft.AspNetCore.Mvc.Razor/TagHelperComponent/ITagHelperComponent.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelperComponent
+{
+    public interface ITagHelperComponent
+    {
+        /// <summary>
+        /// Set to <c>true</c> if <see cref="ITagHelperComponent"/> applies to the given <see cref="TagHelperContext"/>.
+        /// </summary>
+        /// <param name="context">Contains information associated with the current HTML tag.</param>
+        bool AppliesTo(TagHelperContext context);
```

Comment:
Implementations of `ITagHelperComponent` can define which HTML element the component applies to with this method. For instance [this test class](https://github.com/aspnet/Mvc/pull/5966/files#diff-23776229270202af5375b0f5072d11e2R17) applies to the `head` element.